okhttp-idna-mapping-table/src/main/kotlin/okhttp3/internal/idn/MappedRange.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.idn
+
+import okio.ByteString
+
+internal sealed interface MappedRange {
```

Comment:
Looking at the data, there does appear to be a common pattern of contiguous monotonically increasing ranges. Would we gain by compressing these down to an initial value plus offset?

```
0041          ; mapped                 ; 0061          # 1.1  LATIN CAPITAL LETTER A
...
005A          ; mapped                 ; 007A          # 1.1  LATIN CAPITAL LETTER Z
```

Also

```
1D777         ; mapped                 ; 03B8          # 3.1  MATHEMATICAL SANS-SERIF BOLD SMALL THETA
...
1D77B         ; mapped                 ; 03BC          # 3.1  MATHEMATICAL SANS-SERIF BOLD SMALL MU
```

Reply:
Yes! I’ve got mergeAdjacentRanges() for this. It also handles merging ranges that are different in the source table but equivalent for our use.

Reply:
The generated source seem to have A, B, C etc. Is that working?

Reply:
<img width="1000" alt="image" src="https://user-images.githubusercontent.com/231923/235342218-68ee605c-eb81-4461-a716-20fc073072db.png">

Reply:
Ahhhh, I misunderstood your suggestion.

I tried it, and the ranges string shrinks from 32,612 bytes to 30,068 bytes once I limit the mappings to only those where the delta fits in 14 bits.

But by increasing the maximum delta to 18 bits, I can shrink it much more, to 24,756 bytes. To do this, I need to borrow some bits from `b1`, and conveniently there’s some available. I can use the range from 64..118 that was unassigned.

```diff
 0..63 : Length of the UTF-16 sequence that this range maps to. The offset is b2b3.
+ 64..79 : Offset by a fixed negative offset. The bottom 4 bits of the offset are the top 4 bits of the offset.
+ 80..95 : Offset by a fixed positive offset. The bottom 4 bits of the offset are the top 4 bits of the offset.
   119 : Ignored.
   120 : Valid.
   121 : Disallowed
   122 : Mapped inline to the sequence: [b2].
   123 : Mapped inline to the sequence: [b2a].
   124 : Mapped inline to the sequence: [b2, b3].
   125 : Mapped inline to the sequence: [b2a, b3].
   126 : Mapped inline to the sequence: [b2, b3a].
   127 : Mapped inline to the sequence: [b2a, b3a].
```

I’ll hold this for a follow-up.